File: theme-tests/sample.go
package main

import "fmt"

// Go: packages, structs, receivers, constants, fields, and short declarations.
const limit = 3

type Swatch struct {
    Label string
    Color string
}

// Format returns a compact label for one palette entry.
func (swatch Swatch) Format(index int) string {
    return fmt.Sprintf("%d / %s / %s", index, swatch.Label, swatch.Color)
}

func render(swatches []Swatch, enabled bool) []string {
    labels := make([]string, 0, len(swatches))
    if !enabled {
        return labels
    }
    for index, swatch := range swatches {
        if index >= limit {
            break
        }
        labels = append(labels, swatch.Format(index))
    }
    return labels
}

func main() {
    swatches := []Swatch{{Label: "Mint", Color: "#72E0B2"}}
    for _, label := range render(swatches, true) {
        fmt.Println(label)
    }
}

File: theme-tests/sample.rs
use std::fmt;

// Rust 2021: attributes, traits, lifetimes, fields, macros, and pattern matching.
const LIMIT: usize = 3;

#[derive(Debug, Clone)]
struct Swatch<'a> {
    label: &'a str,
    color: &'a str,
}

impl fmt::Display for Swatch<'_> {
    fn fmt(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(formatter, "{} / {}", self.label, self.color)
    }
}

/// Build a bounded list without changing any palette values.
fn render(swatches: &[Swatch<'_>], enabled: bool) -> Vec<String> {
    let mut labels = Vec::new();
    if !enabled {
        return labels;
    }
    for (index, swatch) in swatches.iter().take(LIMIT).enumerate() {
        labels.push(format!("{index} / {swatch}"));
    }
    labels
}

fn main() {
    let swatches = [Swatch { label: "Mint", color: "#72E0B2" }];
    let labels = render(&swatches, true);
    match labels.first() {
        Some(label) => println!("HiAstra: {label}"),
        None => println!("No swatches"),
    }
}
